nrf52-hal-common/src/uarte.rs

```diff
@@ -95,21 +91,16 @@ impl<T> Uarte<T> where T: Instance {
         });
 
         // Enable UARTE instance
-        uarte.enable.write(|w|
-            w.enable().enabled()
-        );
+        uarte.enable.write(|w| w.enable().enabled());
 
         // Configure
         let hardware_flow_control = pins.rts.is_some() && pins.cts.is_some();
-        uarte.config.write(|w|
-            w.hwfc().bit(hardware_flow_control)
-             .parity().variant(parity)
-        );
+        uarte
+            .config
+            .write(|w| w.hwfc().bit(hardware_flow_control).parity().variant(parity));
```

Comment:
Actually I have to disagree. I find the official style much more readable than the previous version.

Reply:
I have to say I'm baffled about this comment. Those svd2rust function calls always come in pairs, so I figured it was self-evident that any formatting choice that makes it easy to immediately recognize what those pairs are would be a readability win.

Could you elaborate what you find more readable about the new style here? I'm genuinely interested.

Reply:
I don't want to hunt for the closure argument on the next line, the merging of the pairs (due to fitting in a single line) I don't mind as much.

If anything I'd have preferred:
```
    uarte.config.write(|w| w
        .hwfc().bit(hardware_flow_control)
        .parity().variant(parity)
    );
```
or the version rustfmt would have generated for a longer version, like
```
    uarte.config.write(|w| {
        w.hwfc()
            .bit(hardware_flow_control)
            .parity()
            .variant(parity)
    });
```
because it highlights exactly the flow of the register manipulation

1. Take the register provided as closure argument
2. Extract the position of the `hwfc` bits
3. Set the bits to `hardware_flow_control`
4. Extract the position of the `parity` bits
5. Set the bits to `parity`

I'm not too hot about the pair argument since they're not expressed in the API, it's implicit knowledge that you have and easy to get wrong, if anything the API should provide setters instead like:
```
    uarte.config.write(|w| w.hwfc(hardware_flow_control).parity(parity));
```

but it doesn't, so the point is moot.

Reply:
> If anything I'd have preferred [...] the version rustfmt would have generated for a longer version [...] because it highlights exactly the flow of the register manipulation

I don't see how that version highlights the flow of register manipulation better than any other. The order of operations is the same in all versions. Only whether they are split over lines, and how, changes.

> I'm not too hot about the pair argument since they're not expressed in the API, it's implicit knowledge that you have and easy to get wrong

I'm not sure I understand what you mean. The API doesn't use the word "pair", but the method that selects a field and the method that sets its value clearly belong together. The first is a method of `w`, the second is a method of the type returned by the first method.

If we ignore how the API is structured, they still clearly belong together. One method sets the value of a field and doesn't make any sense without the method that selects which field that is. That you would prefer to have only one method that does both (your last code example) supports that point.

That this isn't clear to everyone is **exactly** why the formatting should highlight those relationships, not hide them.

> if anything the API should provide setters instead like:

I might prefer that too, but as you say, the point is moot.

Reply:
Isn't the whole point of machine formatted code to avoid random unrelated style changes? If we don't like what the tool does, we should teach the tool to do what we do like (rustfmt.toml, lints, or whatever) and if it is uncapable of being taught, I think we should fix the tool, or live with it.

One possible workaround is to stop chaining all the calls together:

```rust
peripheral.register.write(|w| {
    w.foo.value(1);
    w.bar.value(2);
    unsafe { w.baz.bits(25); }
});
```

Reply:
> Isn't the whole point of machine formatted code to avoid random unrelated style changes?

If that's the case, it's failing spectacularly, since it's doing the exact opposite.

> If we don't like what the tool does, we should teach the tool to do what we do like (rustfmt.toml, lints, or whatever) and if it is uncapable of being taught, I think we should fix the tool

Then please go teach the tool how to understand code beyond the syntax level so it can make sensible decisions. Personally, I don't feel like dropping everything I currently do for a decade or so to make that happen.

> or live with it.

I don't believe it's self-evident that code should be machine-formatted always in every situation. I know it's supposed to be such a universal solution, but that by itself won't make me roll over against my better judgement.

> One possible workaround is to stop chaining all the calls together:

I think that's an excellent suggestion. That still won't make me love rustfmt, but it would take away my biggest gripe.